container-engine-lib/lib/backend_impls/kubernetes/object_attributes_provider/enclave_object_attributes_provider.go

```diff
@@ -368,6 +377,7 @@ func (provider *kubernetesEnclaveObjectAttributesProviderImpl) getLabelsForEncla
 		return nil, stacktrace.Propagate(err, "An error occurred creating a Kubernetes label value from ID string '%v'", id)
 	}
 	labels[kubernetes_label_key.IDKubernetesLabelKey] = idLabelValue
+	labels[kubernetes_label_key.LogsServiceNameKubernetesLabelKey] = idLabelValue
```

Comment:
Something seems wrong here... the function is `EnclaveObject` (so we don't know what type of object it is), but we're attaching a `ServiceName` label (assuming it's a service without knowing)

Reply:
veryy good point, these labels are being attached at the wrong level. I've changed the logic to add these labels only for the user service enclave objects in `ForUserServicePod`